src/core/qgsmaplayer.cpp

```diff
@@ -1980,3 +1990,41 @@ void QgsMapLayer::onNotifiedTriggerRepaint( const QString &message )
   if ( refreshOnNotifyMessage().isEmpty() || refreshOnNotifyMessage() == message )
     triggerRepaint();
 }
+
+QgsRectangle QgsMapLayer::wgs84Extent( bool forceRecalculate ) const
+{
+  QgsRectangle wgs84Extent;
+
+  if ( ! forceRecalculate && ! mWgs84Extent.isNull() )
+  {
+    wgs84Extent = mWgs84Extent;
+  }
+  else
+  {
+    const QgsCoordinateTransform transformer { crs(), QgsCoordinateReferenceSystem::fromOgcWmsCrs( geoEpsgCrsAuthId() ), transformContext() };
+    try
+    {
+      wgs84Extent = transformer.transformBoundingBox( mExtent );
```

Comment:
What if mExtent is null here?

Reply:
Done (in this case the wgs84 extent is null too now)